pageserver/src/tenant.rs

```diff
@@ -2569,12 +2570,13 @@ impl Tenant {
         horizon: u64,
         pitr: Duration,
         ctx: &RequestContext,
+        cancel: &CancellationToken,
     ) -> anyhow::Result<GcResult> {
         let mut totals: GcResult = Default::default();
         let now = Instant::now();
 
         let gc_timelines = self
-            .refresh_gc_info_internal(target_timeline_id, horizon, pitr, ctx)
+            .refresh_gc_info_internal(target_timeline_id, horizon, pitr, ctx, cancel)
             .await?;
```

Comment:
Do we want this to return an error on cancellation? In the loop below, where we check for `task_mgr::is_shutdown_requested()', we return Ok if requested to shut down. Not sure what the right answer is, but seems inconsistent.